### PR TITLE
usm: http2: tests: Cut 9m from TestRawTraffic

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

The PR reuses the same USM instance, and just clears the http2 maps in between the runs, to have a clean state in the next test.

Now each instance of TestRawTraffic (CO-RE, RC, prebuilt) x (TLS, without TLS) takes 20s each (prebuilt TLS is not supported), instead of 100-168s before

### Motivation

Faster CI for system-probe teams

### CODE REVIEW

File: pkg/network/protocols/http2/testutils.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		return
+	}
+
+	for _, mapElement := range Spec.Maps {
+		if strings.HasSuffix(mapElement.Name, "http2_batches") || strings.HasSuffix(mapElement.Name, "http2_telemetry") {
+			continue
+		}
+		mapInstance, _, err := m.GetMap(mapElement.Name)
+		if err != nil {
+			continue
```
Comment (amitslavin) [inline pkg/network/protocols/http2/testutils.go:34]: shouldn't we at least log if we were not able to get the map?

File: pkg/network/protocols/http2/testutils.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+package http2
+
+import (
+	"strings"
+
+	"github.com/cilium/ebpf"
+)
+
+// CleanHTTP2Maps deletes all entries from the http2 maps. Test utility to allow reusing USM instance without carring
```
Comment (amitslavin) [inline pkg/network/protocols/http2/testutils.go:16]:
```suggestion
// CleanHTTP2Maps deletes all entries from the http2 maps. Test utility to allow reusing USM instance without carrying
```

File: pkg/network/protocols/http2/testutils.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		}
+		key := make([]byte, mapInstance.KeySize())
+		value := make([]byte, mapInstance.ValueSize())
+		mapEntries := mapInstance.Iterate()
+		var keysToDelete [][]byte
+		for mapEntries.Next(&key, &value) {
+			keysToDelete = append(keysToDelete, key)
+		}
+		for _, key := range keysToDelete {
+			_ = mapInstance.Delete(&key)
```
Comment (amitslavin) [inline pkg/network/protocols/http2/testutils.go:48]: same question as above, if we were not able to delete for some reason, wouldn't we like at least to log the error?

**Review by amitslavin — APPROVED** [review]

File: pkg/network/protocols/http2/testutils.go (hunk 1 above)
Comment (guyarb) [reply pkg/network/protocols/http2/testutils.go:34]: we can, but I'm not sure it is useful for the context of the test
that's very strange to happen, and unless the test fails, we won't see the log

File: pkg/network/protocols/http2/testutils.go (hunk 2 above)
Comment (guyarb) [reply pkg/network/protocols/http2/testutils.go:16]: actually it is caring :)

File: pkg/network/protocols/http2/testutils.go (hunk 3 above)
Comment (guyarb) [reply pkg/network/protocols/http2/testutils.go:48]: we're doing best effort
if the test passes we won't see the error, so I'm not sure what will it give us

**Pull request merged**